DataFormats/Math/test/cudaAtan2Test.cu

```diff
@@ -59,75 +56,76 @@ __global__ void diffAtan(int * diffs) {
  
-        int diffs[3];
-        cuda::memory::device::zero(diff_d.get(),3*4);
-
-        // Launch the diff CUDA Kernel
-        dim3 threadsPerBlock(32,32,1);
-        dim3 blocksPerGrid ( (Nsteps + threadsPerBlock.x - 1) / threadsPerBlock.x,
-                             (Nsteps + threadsPerBlock.y - 1) / threadsPerBlock.y,
-                            1
-                               );
-        std::cout
-                << "CUDA kernel 'diff' launch with " << blocksPerGrid.x
-                << " blocks of " << threadsPerBlock.y << " threads\n";
-
-        cuda::launch(
-                diffAtan<DEGREE>,
-                { blocksPerGrid, threadsPerBlock },
-                diff_d.get() );
-
-        cuda::memory::copy(diffs, diff_d.get(), 3*4);
-        delta += (std::chrono::high_resolution_clock::now()-start);
+  int diffs[3];
+  cuda::memory::device::zero(diff_d.get(),3*4);
+
+  // Launch the diff CUDA Kernel
+  dim3 threadsPerBlock(32,32,1);
+  dim3 blocksPerGrid((Nsteps + threadsPerBlock.x - 1) / threadsPerBlock.x,
+                     (Nsteps + threadsPerBlock.y - 1) / threadsPerBlock.y,
+                     1);
+  std::cout
+    << "CUDA kernel 'diff' launch with " << blocksPerGrid.x
+    << " blocks of " << threadsPerBlock.y << " threads\n";
+
+  cuda::launch(
+    diffAtan<DEGREE>,
+    { blocksPerGrid, threadsPerBlock },
+    diff_d.get());
+
+  cuda::memory::copy(diffs, diff_d.get(), 3*4);
+  delta += (std::chrono::high_resolution_clock::now()-start);
  
-        float mdiff = diffs[0]*1.e-7;
-        int idiff = diffs[1];
-        int sdiff = diffs[2];
-
-        std::cout << "for degree " << DEGREE << " max diff is " << mdiff 
-                             << ' ' << idiff << ' ' << int2phi(idiff) 
-                             << ' ' << sdiff << ' ' << short2phi(sdiff) <<  std::endl;
-
-
-        std::cout <<"cuda computation took "
-              << std::chrono::duration_cast<std::chrono::milliseconds>(delta).count()
-              << " ms" << std::endl;
-
-	return;
+  float mdiff = diffs[0]*1.e-7;
+  int idiff = diffs[1];
+  int sdiff = diffs[2];
+
+  std::cout << "for degree " << DEGREE << " max diff is " << mdiff 
+            << ' ' << idiff << ' ' << int2phi(idiff) 
+            << ' ' << sdiff << ' ' << short2phi(sdiff) <<  std::endl;
+  std::cout << "cuda computation took "
+            << std::chrono::duration_cast<std::chrono::milliseconds>(delta).count()
+            << " ms" << std::endl;
 }
 
-
 int main() {
-try{
-  go<3>();
-  go<5>();
-  go<7>();
-  go<9>();
-} catch(cuda::runtime_error ex) {
-   std::cout << "cuda error " << ex.what() << std::endl;
-} catch(...) {
-   std::cout <<	"a non cuda error" << std::endl;
-}
-
-
-  return 0;
+  int count = 0;
+  auto status = cudaGetDeviceCount(& count);
+  if (status != cudaSuccess) {
+    std::cerr << "Failed to initialise the CUDA runtime, the test will be skipped." << "\n";
+    exit(EXIT_SUCCESS);
```

Comment:
This is cosmetics, but why `exit` and not `return`?

Reply:
Because I copy-pasted it from the `go` function, and did not change `exit` into `return`.